src/magneticField/UF23Field.cpp

```diff
@@ -0,0 +1,554 @@
+#include "crpropa/magneticField/UF23Field.h"
+
+#include <exception>
+#include <limits>
+#include <string>
+#include <cmath>
+
+// local helper functions and constants
+namespace uf23 {
+
+  template<typename T>
+  crpropa::Vector3d CylToCart(const T v, const double cosPhi, const double sinPhi)
+  {
+    return crpropa::Vector3d(v[0] * cosPhi - v[1] * sinPhi,
+                             v[0] * sinPhi + v[1] * cosPhi,
+                             v[2]);
+  }
+
+  template<typename T>
+  crpropa::Vector3d CartToCyl(const T v, const double cosPhi, const double sinPhi)
+  {
+    return crpropa::Vector3d(v[0] * cosPhi + v[1] * sinPhi,
+                             -v[0] * sinPhi + v[1] * cosPhi,
+                             v[2]);
+  }
+
+  // logistic sigmoid function
+  inline
+  double
+  Sigmoid(const double x, const double x0, const double w)
+  {
+    return 1 / (1 + exp(-(x-x0)/w));
+  }
+
+  // angle between v0 = (cos(phi0), sin(phi0)) and v1 = (cos(phi1), sin(phi1))
+  inline
+  double
+  DeltaPhi(const double phi0, const double phi1)
+  {
+    return acos(cos(phi1)*cos(phi0) + sin(phi1)*sin(phi0));
+  }
+
+  const double kPi = 3.1415926535897932384626;
+  const double kTwoPi = 2*kPi;
+  const double degree = kPi/180.;
+  const double kpc = 1;
+  const double microgauss = 1;
+  const double megayear = 1;
+  const double Gpc = 1e6*kpc;
+  const double pc = 1e-3*kpc;
+  const double second  = megayear / (1e6*60*60*24*365.25);
+  const double kilometer = kpc / 3.0856775807e+16;
+}
+
+namespace crpropa {
+UF23Field::UF23Field(const ModelType mt) :
```

Comment:
Could you add a comment that the UF23 model is internally using 1e-6*gauss=1, kpc=1, etc.

If you have a better place to make this statement I am also fine with it.

Reply:
Added a comment it at this point of cpp file, since it is internal to the code and not visible to the outside user.